modules/pubwiseAnalyticsAdapter.js

```diff
@@ -54,72 +73,268 @@ function enrichWithUTM(dataBag) {
   try {
     for (let prop in utmKeys) {
       utmKeys[prop] = utils.getParameterByName(prop);
-      if (utmKeys[prop] != '') {
+      if (utmKeys[prop]) {
         newUtm = true;
         dataBag[prop] = utmKeys[prop];
       }
     }
 
     if (newUtm === false) {
       for (let prop in utmKeys) {
-        let itemValue = storage.getDataFromLocalStorage(`pw-${prop}`);
-        if (itemValue.length !== 0) {
+        let itemValue = localStorage.getItem(setNamespace(prop));
+        if (itemValue !== null && typeof itemValue !== 'undefined' && itemValue.length !== 0) {
```

Comment:
what about `if (itemValue)` ?

Reply:
I'd prefer not to change that. I'm relatively sure that code is the result of some edge cases.